gpii/node_modules/flowManager/src/SolutionsRegistryDataSource.js

```diff
@@ -35,45 +43,197 @@ fluid.defaults("gpii.flowManager.solutionsRegistry.dataSource", {
     members: {
         fullSolutionsRegistry: null
     },
-    readOnlyGrade: "gpii.flowManager.solutionsRegistry.dataSource",
     invokers: {
-        getImpl: {
-            funcName: "gpii.flowManager.solutionsRegistry.dataSource.handle",
-            args: ["{that}", "{arguments}.1", "{arguments}.2"]
-                             // options, directModel
+        loadFromLocalDisk: {
+            funcName: "gpii.flowManager.solutionsRegistry.dataSource.loadFromLocalDisk",
+            args: ["{that}"]
         }
     },
+    events: {
+        loadSolutions: null,
+        solutionsRegistryReady: null
+    },
     listeners: {
-        onCreate: "gpii.flowManager.solutionsRegistry.dataSource.loadSolutionsRegistry"
+        "onCreate.loadSolutions": {
+            listener: "fluid.promise.fireTransformEvent",
+            args: ["{that}.events.loadSolutions"]
+        }
     }
 });
 
-// TODO: Add an invoker to reload once we are "more live".
-gpii.flowManager.solutionsRegistry.dataSource.loadSolutionsRegistry = function (that) {
+/**
+ * Load the solutions registry from the local file system -- the file system
+ * that this solutions registry data source is running on.
+ *
+ * @param {Object} that - The gpii.flowManager.solutionsRegistry.dataSource.
+ * @param {String} that.options.path - The path to the solutions registry
+ *          directory.
+ * @param {Object} that.fullSolutionsRegistry - This will be set to the
+ *          collection of solution registries given by path.
+ */
+gpii.flowManager.solutionsRegistry.dataSource.loadFromLocalDisk = function (that) {
     if (!that.options.path) {
         fluid.fail("The solutionsRegistry datasource ", that, " needs a \"path\" option pointing to the solution entries folder");
     }
     var url = fluid.module.resolvePath(that.options.path);
     if (!fs.existsSync(url)) {
         fluid.fail("The path provided to the solutionsRegistry datasource (", url, ") has not been found on the file system");
     }
-
     that.fullSolutionsRegistry = fluid.freezeRecursive(require(url));
 };
 
+// Solutions registry data source used by the Cloud Based Flow Manager
+fluid.defaults("gpii.flowManager.solutionsRegistry.dataSource.cloud", {
```

Comment:
We probably should name this component `gpii.flowManager.solutionsRegistry.dataSource.cloudBased` (cloud -> cloudBased) to match the the name of Cloud Based Flow Manager `gpii.flowManager.cloudBased`.

Reply:
Okay